packages/governance/src/multisig/multisig.cairo

```diff
@@ -0,0 +1,503 @@
+                    }
+                    // Remove the last signer
+                    self.Multisig_is_signer.write(signer_to_remove, false);
+                    self.Multisig_signers_by_index.write(last_index, Zero::zero());
+                    self.Multisig_signers_indices.write(signer_to_remove, 0);
+
+                    self.emit(SignerRemoved { signer: signer_to_remove });
+
+                    signers_count -= 1;
+                };
+                self.Multisig_signers_info.write(SignersInfo { quorum, signers_count });
+            }
+            self._change_quorum(new_quorum);
+        }
+
+        fn _replace_signer(
+            ref self: ComponentState<TContractState>,
+            signer_to_remove: ContractAddress,
+            signer_to_add: ContractAddress
+        ) {
+            assert(signer_to_add.is_non_zero(), Errors::ZERO_ADDRESS);
+            assert(!Multisig::is_signer(@self, signer_to_add), Errors::ALREADY_A_SIGNER);
+            assert(Multisig::is_signer(@self, signer_to_remove), Errors::NOT_A_SIGNER);
+
+            self.Multisig_is_signer.write(signer_to_remove, false);
+            self.Multisig_is_signer.write(signer_to_add, true);
+            let index = self.Multisig_signers_indices.read(signer_to_remove);
+            self.Multisig_signers_by_index.write(index, signer_to_add);
+            self.Multisig_signers_indices.write(signer_to_add, index);
+            self.Multisig_signers_indices.write(signer_to_remove, 0);
+
+            self.emit(SignerRemoved { signer: signer_to_remove });
+            self.emit(SignerAdded { signer: signer_to_add });
+        }
+
+        fn _change_quorum(ref self: ComponentState<TContractState>, new_quorum: u32) {
+            let SignersInfo { quorum: old_quorum, signers_count } = self
+                .Multisig_signers_info
+                .read();
+            if new_quorum != old_quorum {
+                assert(new_quorum.is_non_zero(), Errors::ZERO_QUORUM);
+                assert(new_quorum <= signers_count, Errors::QUORUM_TOO_HIGH);
+                self.Multisig_signers_info.write(SignersInfo { quorum: new_quorum, signers_count });
+                self.emit(QuorumUpdated { old_quorum, new_quorum });
+            }
+        }
+    }
+}
+
+//
+// Storage helpers
+//
+
+const _2_POW_32: NonZero<u128> = 0xFFFFFFFF;
+const _2_POW_64: NonZero<u128> = 0xFFFFFFFFFFFFFFFF;
+
+#[derive(Drop)]
+struct TxInfo {
+    is_executed: bool,
+    submitted_block: u64
+}
+
+impl TxInfoStorePacking of StorePacking<TxInfo, u128> {
+    fn pack(value: TxInfo) -> u128 {
+        let TxInfo { is_executed, submitted_block } = value;
+        let is_executed_value = if is_executed {
+            1
+        } else {
+            0
+        };
+        is_executed_value * _2_POW_64.into() + submitted_block.into()
+    }
+
+    fn unpack(value: u128) -> TxInfo {
+        let (is_executed_value, submitted_block) = u128_safe_divmod(value, _2_POW_64);
```

Comment:
If we store the boolean value at the end, we can divide by 2 which is probably more efficient.

Reply:
Improved the packing logic by shifting `submitted_block` only 1 bit left and storing `is_executed` value as the last bit